OutSystems/outsystems-datagrid-reactive
# Rou 3142

This PR adds a new API called **MarkChangesAsSavedByKey**, to mark a group of Data Grid rows with given keys (from the KeyBinding field) as saved in the database.

### Test Steps

**Validate client action MarkChangesAsSavedByKey with success scenario:**
1. Open Page
2. Add a new row by clicking in "Add New Row"
3. For the first row, change the content of column Product Name and a dirty mark will appear
4. Fill in the Row Keys values of "-1,45"
4. Press "Mark Changes As Saved By Key" button

Expected: 
- A success message should appear "Mark Changes As Saved successfully";
- No dirty mark is shown



**Validate client action MarkChangesAsSavedByKey with success scenario - Force Clean Invalids:**
1. Open Page
2. Add a new row by clicking in "Add New Row"
3. Click in "Set Cell Invalid"
4. Enable "Force Clean Invalids"
4. Fill in the Row Keys values of "-1,56"
4. Press "Mark Changes As Saved By Key" button

Expected: 
- A success message should appear "Mark Changes As Saved successfully";
- No dirty mark nor validation marks are shown



**Validate client action MarkChangesAsSavedByKey with empty rows list:**
1. Open Page
2. Press "Mark Changes As Saved By Key - Empty" button

Expected: 
- An error message should appear "Rows list is empty";



**Validate client action MarkChangesAsSavedByKey with rows list containing empty key values:**
1. Open Page
2. Press "Mark Changes As Saved By Key - With Empty Key" button

Expected: 
- An error message should appear "Rows list has empty values.";



**Validate client action MarkChangesAsSavedByKey with invalid row:**
1. Open Page
2. Press "Mark Changes As Saved By Key - Invalid Row" button

Expected: 
- An error message should appear "Row not found.";








### Checklist
* [ X] tested locally
* [ X] documented the code
* [ X] clean all warnings and errors of eslint
* [ X] requires changes in OutSystems (if so, provide a module with changes)
* [ X] requires new sample page in OutSystems (if so, provide a module with changes)
